akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -0,0 +1,43 @@
+package akka.pattern
+
+import akka.actor.Scheduler
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.control.NonFatal
+
+/**
+ * This trait provides the retry utility function
+ */
+trait RetrySupport {
+
+  /**
+   * Given a function from Unit to Future, returns an internally retrying Future
+   * The first attempt will be made immediately, each subsequent attempt will be made after 'delay'
+   * A scheduler (eg context.system.scheduler) must be provided to delay each retry
+   * If attempts are exhausted the returned future is simply the result of invoking attempt
+   *
+   * <b>Example usage:</b>
+   *
+   * {{{
+   * protected val sendAndReceive: HttpRequest => Future[HttpResponse]
+   * private val sendReceiveRetry: HttpRequest => Future[HttpResponse] = (req: HttpRequest) => retry[HttpResponse](
+   *   attempt = () => sendAndReceive(req),
+   *   attempts = 10,
+   *   delay = 2 seconds,
+   *   scheduler = context.system.scheduler
+   * )
+   * }}}
+   */
+  def retry[T](attempt: () ⇒ Future[T], attempts: Int, delay: FiniteDuration, scheduler: Scheduler)(implicit ec: ExecutionContext): Future[T] = {
+    if (attempts > 0) {
+      attempt() recoverWith {
+        case NonFatal(_) ⇒ after(delay, scheduler) {
+          retry(attempt, attempts - 1, delay, scheduler)
+        }
+      }
+    } else {
+      attempt()
+    }
+  }
```

Comment:
Could we also ship with a backoff based one right away?
`retryWithBackoff`?

Would be similar to https://doc.akka.io/docs/akka/2.5/scala/general/supervision.html#backoff-supervisor

Reply:
I'll be happy to, what do you think bout doing it in a separate PR?

Reply:
Yeah absolutely :)